laws/shared/src/test/scala/algebra/laws/LawTests.scala

```diff
@@ -33,16 +33,23 @@ trait LawTestsBase extends FunSuite with Discipline {
   }
 
   private[laws] def laws[L[_] <: Laws, A](implicit
+      lws: L[A], tag: TypeTagM[A]): LawChecker[L[A]] = laws[L, A]("")
+
+  private[laws] def laws[L[_] <: Laws, A](extraTag: String)(implicit
       laws: L[A], tag: TypeTagM[A]): LawChecker[L[A]] =
-    LawChecker("[" + tag.tpe.toString + "]", laws)
+    LawChecker("[" + tag.tpe.toString + (if(extraTag != "") "@@" + extraTag else "") + "]", laws)
 
   laws[OrderLaws, Boolean].check(_.order)
   laws[LogicLaws, Boolean].check(_.bool)
   laws[LogicLaws, SimpleHeyting].check(_.heyting)
   laws[LatticePartialOrderLaws, Boolean].check(_.boundedLatticePartialOrder)
+  laws[RingLaws, Boolean].check(_.boolRing(BooleanRing))
+
+  // ensure that Bool[A].asCommutativeRing is a valid BoolRing
+  laws[RingLaws, Boolean]("ring-from-bool").check(_.boolRing(Bool[Boolean].asCommutativeRing))
```

Comment:
Since we now have two `RuleSet`s for `[RingLaws, Boolean]`, it was necessary to somehow distinguish the property names. This is done by an extra String tag `ring-from-bool`.

Similarly below for `[LogicLaws, Boolean]`.